app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostFragment.kt

```diff
@@ -250,7 +268,11 @@ class ZimHostFragment : BaseFragment(), ZimHostCallbacks, ZimHostContract.View {
 
   override fun onResume() {
     super.onResume()
-    presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
+    if (!ServerUtils.isServerStarted) {
+      presenter.loadBooks(sharedPreferenceUtil.allBooksToHost)
+    } else {
+      presenter.loadBooks(sharedPreferenceUtil.hostedBooks)
```

Comment:
@macgills  I'm stuck here. It doesn't work as I expected.

Reply:
you don't need to change this, look to `ZimHostPresenter`. It loads all the books and sets whatever you pass in to `loadBooks` as selected

Reply:
If we go to the mainActivity and then if we return to the `ZimHostActivity` or if lock the screen it won't appear in the recyclerview.

Reply:
just have the loadBooks call, whenever this resumes it should refresh from the database so our data is up to date. If it won't appear in the recyclerview then find out why, what is returning empty or setting the items to empty